server/controllers/journal.js

```diff
@@ -1104,57 +1105,72 @@ function handleCreditNote (id, user_id, done) {
+  })
+  .then(function (results) {
     queries.items = [];
-    data.forEach(function (item) {
-      // Debit sale items
-      // var itemSql =
-      // 'INSERT INTO `posting_journal` ' +
-      //   '(`project_id`, `fiscal_year_id`, `period_id`, `trans_id`, `trans_date`, ' +
-      //   '`description`, `account_id`, `credit`, `debit`, `credit_equiv`, `debit_equiv`, ' +
-      //   '`currency_id`, `deb_cred_uuid`, `deb_cred_type`, `inv_po_id`, `origin_id`, `user_id` ) ' +
-      // 'SELECT `sale`.`project_id`, ' + [fiscalYearId, periodId, transId, '\'' + get.date() + '\''].join(', ') + ', ' +
-      //   '\'' + reference_note.description + '', `inventory_group`.`sales_account`, `sale_item`.`debit`, `sale_item`.`credit`, ' +
-      //   '`sale_item`.`debit`, `sale_item`.`credit`, `sale`.`currency_id`, null, ' +
-      //   ' null, `sale`.`uuid`, ' + [originId, user_id].join(', ') + ' ' +
-      // 'FROM `sale` JOIN `sale_item` JOIN `inventory` JOIN `inventory_group` ON ' +
-      //   '`sale_item`.`sale_uuid`=`sale`.`uuid` AND `sale_item`.`inventory_uuid`=`inventory`.`uuid` AND ' +
-      //   '`inventory`.`group_uuid`=`inventory_group`.`uuid` ' +
-      // 'WHERE `sale`.`uuid`=' + sanitize.escape(reference.sale_uuid) + ';';
-
-      var itemSql =
-        'INSERT INTO `posting_journal` ' +
-          '(`project_id`, `uuid`, `fiscal_year_id`, `period_id`, `trans_id`, `trans_date`, ' +
-          '`description`, `account_id`, `credit`, `debit`, `credit_equiv`, `debit_equiv`, ' +
-          '`currency_id`, `deb_cred_uuid`, `deb_cred_type`, `inv_po_id`, `origin_id`, `user_id` ) ' +
-        'SELECT `sale`.`project_id`, ' + [sanitize.escape(uuid()), cfg.fiscalYearId, cfg.periodId, transId, '\'' + get.date() + '\''].join(', ') + ', ' +
-          '\'' + reference.description + '\', `inventory_group`.`sales_account`, `sale_item`.`debit`, `sale_item`.`credit`, ' +
-          '`sale_item`.`debit`, `sale_item`.`credit`, `sale`.`currency_id`, null, ' +
-          ' null, `sale`.`uuid`, ' + [cfg.originId, user_id].join(', ') + ' ' +
-        'FROM `sale` JOIN `sale_item` JOIN `inventory` JOIN `inventory_group` ON ' +
-          '`sale_item`.`sale_uuid`=`sale`.`uuid` AND `sale_item`.`inventory_uuid`=`inventory`.`uuid` AND ' +
-          '`inventory`.`group_uuid`=`inventory_group`.`uuid` ' +
-        'WHERE `sale_item`.`uuid`=' + sanitize.escape(item.item_uuid) + ';';
+    var date = get.date();
 
-      queries.items.push(itemSql);
-    });
+    results.forEach(function (item) {
+      item.uuid = sanitize.escape(uuid());
+      item.origin_id = cfg.originId;
+      item.description = cfg.descrip;
+      item.period_id = cfg.periodId;
+      item.fiscal_year_id = cfg.fiscalYearId;
+      item.trans_id = cfg.trans_id;
+      item.trans_date = util.toMysqlDate(get.date());
 
-    return db.exec(queries.debtor);
-  })
-  .then(function () {
+      //FIX ME deb_cred_uuid when is empty the text 'null' is inserted in the table posting_journal
```

Comment:
If you use `db.exec(sql, [param1, param2, ...])` instead of sanitize.escape, you can db.js will replace a the javascript object `null` will the a NULL database value.  For more information, see this [link](https://github.com/felixge/node-mysql/#escaping-query-values)